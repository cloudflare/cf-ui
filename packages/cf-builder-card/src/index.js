import cardActions from './cardActions';
import CardActionTypes from './CardActionTypes';
import CardBuilder from './CardBuilder';
import cardReducer from './cardReducer';

export { cardActions, CardActionTypes, CardBuilder, cardReducer };
