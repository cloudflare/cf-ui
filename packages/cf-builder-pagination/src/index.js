import PaginationBuilder from './PaginationBuilder';

export { PaginationBuilder };
