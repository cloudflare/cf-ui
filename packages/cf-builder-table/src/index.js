import tableActions from './tableActions';
import TableActionTypes from './TableActionTypes';
import TableBuilder from './TableBuilder';
import TableBuilderPropTypes from './TableBuilderPropTypes';
import tableReducer from './tableReducer';

export {
  tableActions,
  TableActionTypes,
  TableBuilder,
  TableBuilderPropTypes,
  tableReducer
};
