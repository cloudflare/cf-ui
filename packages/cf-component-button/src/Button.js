import React, { PropTypes } from 'react';
import { createComponent } from 'cf-style-container';

const getRadius = (borderRadius, group) => {
  if (group) {
    return {
      borderTopRightRadius: group === 'last' ? borderRadius : 0,
      borderTopLeftRadius: group === 'first' ? borderRadius : 0,
      borderBottomLeftRadius: group === 'first' ? borderRadius : 0,
      borderBottomRightRadius: group === 'last' ? borderRadius : 0
    };
  }
  return { borderRadius };
};

const getOpacity = props => {
  if (props.loading) {
    return 0.8;
  }
  if (props.disabled) {
    return 0.5;
  }
  return 'inherit';
};

const getLoadingBefore = (fadeZoomIn, loading) => {
  if (!loading) {
    return null;
  }
  // this needs something better (e.g. a rotating icon)
  return {
    '&::before': {
      content: '"â€¦"',
      color: 'black',
      display: 'block',
      position: 'absolute',
      top: '0.6em',
      left: 0,
      height: '100%',
      width: '100%',
      opacity: 1
    }
  };
};

const styles = props => {
  const theme = props.theme;
  return Object.assign(
    {
      '&:hover': {
        backgroundColor: props.loading
          ? theme.disabledBackground
          : theme[`${props.type}HoverBackground`],
        borderColor: theme[`${props.type}HoverBorder`],
        color: props.loading
          ? theme.disabledBackground
          : theme[`${props.type}HoverColor`]
      },
      '&:active': {
        backgroundColor: theme[`${props.type}ActiveBackground`],
        borderColor: theme[`${props.type}ActiveBorder`],
        color: theme[`${props.type}ActiveColor`]
      },
      '&:focus': {
        backgroundColor: props.loading
          ? theme.backgroundColor
          : theme[`${props.type}Background`],
        boxShadow: `inset 0px 0px 0px ${theme.borderSize} ${theme[`${props.type}focusOutlineColor`]}`,
        color: props.loading
          ? theme.disabledBackground
          : theme[`${props.type}FocusColor`],
        outline: props.loading ? 'none' : 'inherit'
      },
      '&[title]': {
        pointerEvents: props.disabled ? 'auto' : 'none'
      },
      backgroundColor: props.loading
        ? theme.disabledBackground
        : theme[`${props.type}Background`],
      borderBottom: theme.borderBottom,
      borderBottomWidth: theme.borderBottomWidth,
      borderColor: theme[`${props.type}Border`],
      borderLeft: props.group && props.group !== 'first' ? 0 : theme.borderLeft,
      borderRight: theme.borderRight,
      borderStyle: theme.borderStyle,
      borderTop: theme.borderTop,
      borderWidth: theme.borderWidth,
      color: props.loading
        ? theme.disabledBackground
        : theme[`${props.type}Color`],
      cursor: props.disabled || props.loading ? 'default' : theme.cursor,
      display: theme.display,
      fontFamily: theme.fontFamily,
      fontSize: theme.fontSize,
      fontWeight: theme.fontWeight,
      lineHeight: theme.lineHeight,
      marginBottom: props.group ? 0 : theme.marginBottom,
      marginLeft: props.group ? 0 : theme.marginLeft,
      marginRight: props.group ? 0 : theme.marginRight,
      marginTop: props.group ? 0 : theme.marginTop,
      opacity: getOpacity(props),
      paddingBottom: theme.paddingBottom,
      paddingLeft: theme.paddingLeft,
      paddingRight: theme.paddingRight,
      paddingTop: theme.paddingTop,
      pointerEvents: props.disabled ? 'none' : 'auto',
      position: theme.position,
      textAlign: theme.textAlign,
      transition: theme.transition,
      userSelect: theme.userSelect
    },
    getRadius(theme.borderRadius, props.group),
    getLoadingBefore(theme.fadeZoomIn, props.loading)
  );
};

const Button = (
  {
    onClick,
    submit,
    className,
    disabled,
    loading,
    children
  }
) => (
  <button
    type={submit ? 'submit' : 'button'}
    disabled={disabled || loading}
    className={className}
    onClick={onClick}
  >
    {children}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  submit: PropTypes.bool,
  className: PropTypes.string.isRequired,
  group: PropTypes.oneOf(['first', 'inbetween', 'last']),
  type: PropTypes.oneOf([
    'default',
    'primary',
    'success',
    'warning',
    'danger'
  ]).isRequired,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  children: PropTypes.node
};

Button.defaultProps = {
  type: 'default',
  submit: false
};

export default createComponent(styles, Button);
