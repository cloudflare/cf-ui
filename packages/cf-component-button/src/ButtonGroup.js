import React, {PropTypes} from 'react';
import {createComponent} from 'cf-style-container';

const styles = props => {
  const theme = props.theme;
  return ({
    display: theme.display,
    position: theme.position
  });
};

const getGroupByIndex = (index, length) => {
  if (index === length - 1) {
    return 'last';
  }
  if (index === 0) {
    return 'first';
  }
  return 'inbetween';
};

const addGroupProps = children =>
  React.Children.map(children, (child, index) =>
    React.cloneElement(child, {
      group: getGroupByIndex(index, React.Children.count(children))
    })
  );

const ButtonGroup = ({children, className}) =>
  <div className={className}>{addGroupProps(children)}</div>;

ButtonGroup.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string.isRequired
};

export default createComponent(styles, ButtonGroup);
