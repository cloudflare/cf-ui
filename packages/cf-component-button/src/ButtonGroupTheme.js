export default {
  display: 'inline-block',
  position: 'relative'
};
