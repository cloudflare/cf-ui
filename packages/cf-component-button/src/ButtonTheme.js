import { darken, lighten } from 'polished';

export default baseTheme => {
  const defaultBorder = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorGrayLight
  );
  const primaryBorder = darken(baseTheme.colorOffsetDark, baseTheme.colorBlue);
  const successBorder = darken(baseTheme.colorOffsetDark, baseTheme.colorGreen);
  const warningBorder = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorOrange
  );
  const dangerBorder = darken(baseTheme.colorOffsetDark, baseTheme.colorRed);
  const defaultHoverBackground = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorGrayLight
  );
  const defaultActiveBackground = darken(
    baseTheme.colorOffsetDark,
    defaultHoverBackground
  );
  const primaryActiveBackground = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorBlueDark
  );
  const successActiveBackground = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorGreenDark
  );
  const warningActiveBackground = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorOrangeDark
  );
  const dangerActiveBackground = darken(
    baseTheme.colorOffsetDark,
    baseTheme.colorRedDark
  );

  return {
    fontSize: `${(parseFloat(baseTheme.fontSize) - 1) / parseFloat(baseTheme.fontSize)}rem`,
    fontWeight: baseTheme.weightNormal,
    paddingTop: '0.6em',
    paddingRight: '1em',
    paddingBottom: '0.53em',
    paddingLeft: '1em',
    spacing: '0.4rem',
    borderSize: '1px',
    display: 'inline-block',
    position: 'relative',
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
    background: 'transparent',
    lineHeight: 1.2,
    textAlign: 'center',
    webkitTextStroke: 0, // This makes white text in buttons hard to read, so removing it
    webkitFontSmoothing: 'subpixel-antialiased',
    userSelect: 'none',
    cursor: 'pointer',
    transition: 'all 200ms ease',
    border: 'none',
    borderLeft: 0,
    borderBottom: '1px solid',
    borderTop: 0,
    borderRight: 0,

    // Disabled
    disabledBorder: darken(
      baseTheme.colorOffsetDark,
      baseTheme.disabledBackground
    ),
    disabledColor: lighten(0.3, baseTheme.disabledBackground),

    // Normal
    defaultBackground: baseTheme.colorGrayLight,
    primaryBackground: baseTheme.colorBlue,
    successBackground: baseTheme.colorGreen,
    warningBackground: baseTheme.colorOrange,
    dangerBackground: baseTheme.colorRed,

    defaultBorder,
    primaryBorder,
    successBorder,
    warningBorder,
    dangerBorder,

    defaultColor: baseTheme.colorGrayDark,
    primaryColor: baseTheme.colorWhite,
    successColor: baseTheme.colorWhite,
    warningColor: baseTheme.colorWhite,
    dangerColor: baseTheme.colorWhite,

    // Hover
    defaultHoverBackground,
    primaryHoverBackground: baseTheme.colorBlueDark,
    successHoverBackground: baseTheme.colorGreenDark,
    warningHoverBackground: baseTheme.colorOrangeDark,
    dangerHoverBackground: baseTheme.colorRedDark,

    defaultHoverBorder: darken(baseTheme.colorOffsetDark, defaultBorder),
    primaryHoverBorder: darken(baseTheme.colorOffsetDark, primaryBorder),
    successHoverBorder: darken(
      baseTheme.colorOffsetDark,
      baseTheme.colorGreenDark
    ),
    warningHoverBorder: darken(
      baseTheme.colorOffsetDark,
      baseTheme.colorOrangeDark
    ),
    dangerHoverBorder: darken(
      baseTheme.colorOffsetDark,
      baseTheme.colorRedDark
    ),

    defaultHoverColor: baseTheme.colorGrayDark,
    primaryHoverColor: baseTheme.colorWhite,
    successHoverColor: baseTheme.colorWhite,
    warningHoverColor: baseTheme.colorWhite,
    dangerHoverColor: baseTheme.colorWhite,

    // Active
    defaultActiveBackground,
    primaryActiveBackground,
    successActiveBackground,
    warningActiveBackground,
    dangerActiveBackground,

    defaultActiveBorder: darken(
      baseTheme.colorOffsetDark,
      defaultActiveBackground
    ),
    primaryActiveBorder: darken(
      baseTheme.colorOffsetDark,
      primaryActiveBackground
    ),
    successActiveBorder: darken(
      baseTheme.colorOffsetDark,
      successActiveBackground
    ),
    warningActiveBorder: darken(
      baseTheme.colorOffsetDark,
      warningActiveBackground
    ),
    dangerActiveBorder: darken(
      baseTheme.colorOffsetDark,
      dangerActiveBackground
    ),

    defaultActiveColor: baseTheme.colorGrayDark,
    primaryActiveColor: baseTheme.colorWhite,
    successActiveColor: baseTheme.colorWhite,
    warningActiveColor: baseTheme.colorWhite,
    dangerActiveColor: baseTheme.colorWhite,

    // Focus
    defaultFocusColor: baseTheme.colorGrayDark,
    primaryFocusColor: baseTheme.colorWhite,
    successFocusColor: baseTheme.colorWhite,
    warningFocusColor: baseTheme.colorWhite,
    dangerFocusColor: baseTheme.colorWhite,

    defaultFocusOutlineColor: darken(
      baseTheme.colorOffsetDark,
      defaultActiveBackground
    ),
    primaryFocusOutlineColor: darken(
      baseTheme.colorOffsetDark,
      primaryActiveBackground
    ),
    successFocusOutlineColor: darken(
      baseTheme.colorOffsetDark,
      successActiveBackground
    ),
    warningFocusOutlineColor: darken(
      baseTheme.colorOffsetDark,
      warningActiveBackground
    ),
    dangerFocusOutlineColor: darken(
      baseTheme.colorOffsetDark,
      dangerActiveBackground
    )
  };
};
