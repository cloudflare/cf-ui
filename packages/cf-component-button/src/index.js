import Button from './Button';
import ButtonTheme from './ButtonTheme';
import ButtonGroup from './ButtonGroup';
import ButtonGroupTheme from './ButtonGroupTheme';

export { Button, ButtonTheme, ButtonGroup, ButtonGroupTheme };
