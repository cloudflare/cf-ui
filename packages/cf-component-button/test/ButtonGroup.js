import React from 'react';
import renderer from 'react-test-renderer';
import felaTestContext from '../../../felaTestContext';
import { applyTheme } from '../../cf-style-container/src/index';
import {
  ButtonGroup as ButtonGroupUnstyled,
  ButtonGroupTheme,
  Button as ButtonUnstyled,
  ButtonTheme
} from '../../cf-component-button/src/index';

const Button = applyTheme(ButtonUnstyled, ButtonTheme);
const ButtonGroup = applyTheme(ButtonGroupUnstyled, ButtonGroupTheme);

test('should render', () => {
  const component = renderer.create(
    felaTestContext(<ButtonGroup>Hello</ButtonGroup>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render 1 button', () => {
  const component = renderer.create(
    felaTestContext(
      <ButtonGroup>
        <Button type="primary" onClick={() => {}}>Button</Button>
      </ButtonGroup>
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render 2 buttons', () => {
  const component = renderer.create(
    felaTestContext(
      <ButtonGroup>
        <Button type="primary" onClick={() => {}}>Button</Button>
        <Button type="primary" onClick={() => {}}>Button</Button>
      </ButtonGroup>
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render 3 buttons', () => {
  const component = renderer.create(
    felaTestContext(
      <ButtonGroup>
        <Button type="primary" onClick={() => {}}>Button</Button>
        <Button type="primary" onClick={() => {}}>Button</Button>
        <Button type="primary" onClick={() => {}}>Button</Button>
      </ButtonGroup>
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});
