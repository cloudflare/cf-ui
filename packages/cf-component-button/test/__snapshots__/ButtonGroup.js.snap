// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="a b c d"
>
  Hello
</div>
`;

exports[`should render 1 button 1`] = `
<div
  className="a b c d"
>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b al am an ao ap aq ar as at au"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`should render 1 button 2`] = `
"
.a {
  display: inline-block
}

.b {
  position: relative
}

.c {
  vertical-align: middle
}

.d {
  white-space: nowrap
}

.e:hover {
  background-color: #2869a2
}

.f:hover {
  border-color: #215685
}

.g:hover {
  color: #fff
}

.h:active {
  background-color: #215685
}

.i:active {
  border-color: #1a4468
}

.j:active {
  color: #fff
}

.k:focus {
  background-color: #2f7bbf
}

.l:focus {
  color: #fff
}

.m:focus {
  border-color: #2869a2
}

.n[title] {
  pointer-events: none
}

.o {
  background-color: #2f7bbf
}

.p {
  border-color: #2869a2
}

.q {
  border-bottom: 1px solid
}

.r {
  border-left: 0px
}

.s {
  border-right: 0px
}

.t {
  border-top: 0px
}

.u {
  color: #fff
}

.v {
  cursor: pointer
}

.w {
  font-family: \\"Open Sans\\", Helvetica, Arial, sans-serif
}

.x {
  font-size: 0.9333333333333333rem
}

.y {
  font-weight: 400
}

.z {
  line-height: 1.2
}

.ab {
  margin-bottom: 0px
}

.ac {
  margin-right: 0px
}

.ad {
  margin-top: 0px
}

.ae {
  margin-left: 0px
}

.af {
  opacity: inherit
}

.ag {
  padding-bottom: 0.53em
}

.ah {
  padding-left: 1em
}

.ai {
  padding-right: 1em
}

.aj {
  padding-top: 0.6em
}

.ak {
  pointer-events: auto
}

.al {
  border-top-right-radius: 2px
}

.am {
  border-top-left-radius: 0px
}

.an {
  border-bottom-left-radius: 0px
}

.ao {
  border-bottom-right-radius: 2px
}

.ap {
  text-align: center
}

.aq {
  transition: all 200ms ease;
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease
}

.ar {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none
}

.as {
  float: none
}

.at {
  max-width: initial
}

.au {
  width: auto
}
"
`;

exports[`should render 2`] = `
"
.a {
  display: inline-block
}

.b {
  position: relative
}

.c {
  vertical-align: middle
}

.d {
  white-space: nowrap
}
"
`;

exports[`should render 2 buttons 1`] = `
<div
  className="a b c d"
>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b al am an ao ap aq ar as at au"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b av aw ax ay ap aq ar as at au"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`should render 2 buttons 2`] = `
"
.a {
  display: inline-block
}

.b {
  position: relative
}

.c {
  vertical-align: middle
}

.d {
  white-space: nowrap
}

.e:hover {
  background-color: #2869a2
}

.f:hover {
  border-color: #215685
}

.g:hover {
  color: #fff
}

.h:active {
  background-color: #215685
}

.i:active {
  border-color: #1a4468
}

.j:active {
  color: #fff
}

.k:focus {
  background-color: #2f7bbf
}

.l:focus {
  color: #fff
}

.m:focus {
  border-color: #2869a2
}

.n[title] {
  pointer-events: none
}

.o {
  background-color: #2f7bbf
}

.p {
  border-color: #2869a2
}

.q {
  border-bottom: 1px solid
}

.r {
  border-left: 0px
}

.s {
  border-right: 0px
}

.t {
  border-top: 0px
}

.u {
  color: #fff
}

.v {
  cursor: pointer
}

.w {
  font-family: \\"Open Sans\\", Helvetica, Arial, sans-serif
}

.x {
  font-size: 0.9333333333333333rem
}

.y {
  font-weight: 400
}

.z {
  line-height: 1.2
}

.ab {
  margin-bottom: 0px
}

.ac {
  margin-right: 0px
}

.ad {
  margin-top: 0px
}

.ae {
  margin-left: 0px
}

.af {
  opacity: inherit
}

.ag {
  padding-bottom: 0.53em
}

.ah {
  padding-left: 1em
}

.ai {
  padding-right: 1em
}

.aj {
  padding-top: 0.6em
}

.ak {
  pointer-events: auto
}

.al {
  border-top-right-radius: 0px
}

.am {
  border-top-left-radius: 2px
}

.an {
  border-bottom-left-radius: 2px
}

.ao {
  border-bottom-right-radius: 0px
}

.ap {
  text-align: center
}

.aq {
  transition: all 200ms ease;
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease
}

.ar {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none
}

.as {
  float: none
}

.at {
  max-width: initial
}

.au {
  width: auto
}

.av {
  border-top-right-radius: 2px
}

.aw {
  border-top-left-radius: 0px
}

.ax {
  border-bottom-left-radius: 0px
}

.ay {
  border-bottom-right-radius: 2px
}
"
`;

exports[`should render 3 buttons 1`] = `
<div
  className="a b c d"
>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b al am an ao ap aq ar as at au"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b al av aw ao ap aq ar as at au"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b ax av aw ay ap aq ar as at au"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`should render 3 buttons 2`] = `
"
.a {
  display: inline-block
}

.b {
  position: relative
}

.c {
  vertical-align: middle
}

.d {
  white-space: nowrap
}

.e:hover {
  background-color: #2869a2
}

.f:hover {
  border-color: #215685
}

.g:hover {
  color: #fff
}

.h:active {
  background-color: #215685
}

.i:active {
  border-color: #1a4468
}

.j:active {
  color: #fff
}

.k:focus {
  background-color: #2f7bbf
}

.l:focus {
  color: #fff
}

.m:focus {
  border-color: #2869a2
}

.n[title] {
  pointer-events: none
}

.o {
  background-color: #2f7bbf
}

.p {
  border-color: #2869a2
}

.q {
  border-bottom: 1px solid
}

.r {
  border-left: 0px
}

.s {
  border-right: 0px
}

.t {
  border-top: 0px
}

.u {
  color: #fff
}

.v {
  cursor: pointer
}

.w {
  font-family: \\"Open Sans\\", Helvetica, Arial, sans-serif
}

.x {
  font-size: 0.9333333333333333rem
}

.y {
  font-weight: 400
}

.z {
  line-height: 1.2
}

.ab {
  margin-bottom: 0px
}

.ac {
  margin-right: 0px
}

.ad {
  margin-top: 0px
}

.ae {
  margin-left: 0px
}

.af {
  opacity: inherit
}

.ag {
  padding-bottom: 0.53em
}

.ah {
  padding-left: 1em
}

.ai {
  padding-right: 1em
}

.aj {
  padding-top: 0.6em
}

.ak {
  pointer-events: auto
}

.al {
  border-top-right-radius: 0px
}

.am {
  border-top-left-radius: 2px
}

.an {
  border-bottom-left-radius: 2px
}

.ao {
  border-bottom-right-radius: 0px
}

.ap {
  text-align: center
}

.aq {
  transition: all 200ms ease;
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease
}

.ar {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none
}

.as {
  float: none
}

.at {
  max-width: initial
}

.au {
  width: auto
}

.av {
  border-top-left-radius: 0px
}

.aw {
  border-bottom-left-radius: 0px
}

.ax {
  border-top-right-radius: 2px
}

.ay {
  border-bottom-right-radius: 2px
}
"
`;

exports[`should render 3 buttons with spacing 1`] = `
<div
  className="a b c d"
>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad ae af ag ah ai aj ak b al am an ao ap aq ar"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad as af ag ah ai aj ak b al am an ao ap aq ar"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
  <button
    className="e f g h i j k l m n o p q r s t u v a w x y z ab ac ad as af ag ah ai aj ak b al am an ao ap aq ar"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`should render 3 buttons with spacing 2`] = `
"
.a {
  display: inline-block
}

.b {
  position: relative
}

.c {
  vertical-align: middle
}

.d {
  white-space: nowrap
}

.e:hover {
  background-color: #2869a2
}

.f:hover {
  border-color: #215685
}

.g:hover {
  color: #fff
}

.h:active {
  background-color: #215685
}

.i:active {
  border-color: #1a4468
}

.j:active {
  color: #fff
}

.k:focus {
  background-color: #2f7bbf
}

.l:focus {
  color: #fff
}

.m:focus {
  border-color: #2869a2
}

.n[title] {
  pointer-events: none
}

.o {
  background-color: #2f7bbf
}

.p {
  border-color: #2869a2
}

.q {
  border-bottom: 1px solid
}

.r {
  border-left: 0px
}

.s {
  border-right: 0px
}

.t {
  border-top: 0px
}

.u {
  color: #fff
}

.v {
  cursor: pointer
}

.w {
  font-family: \\"Open Sans\\", Helvetica, Arial, sans-serif
}

.x {
  font-size: 0.9333333333333333rem
}

.y {
  font-weight: 400
}

.z {
  line-height: 1.2
}

.ab {
  margin-bottom: 0px
}

.ac {
  margin-right: 0px
}

.ad {
  margin-top: 0px
}

.ae {
  margin-left: 0px
}

.af {
  opacity: inherit
}

.ag {
  padding-bottom: 0.53em
}

.ah {
  padding-left: 1em
}

.ai {
  padding-right: 1em
}

.aj {
  padding-top: 0.6em
}

.ak {
  pointer-events: auto
}

.al {
  border-radius: 2px
}

.am {
  text-align: center
}

.an {
  transition: all 200ms ease;
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease
}

.ao {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none
}

.ap {
  float: none
}

.aq {
  max-width: initial
}

.ar {
  width: auto
}

.as {
  margin-left: 0.4rem
}
"
`;
